File: Semana2/Clase3/Perro/src/Perro.java
class Perro {
    public String nombre;
    public String raza;
    public int edad;
    public float altura;
    public char sexo;// 'M'->Macho, 'H'->Hembra, 'I'->Indeterminado
    public boolean dormir = false;

    // constructor//
    public Perro(String nombre, String raza, int edad, float altura, char sexo) {
        this.nombre = nombre;
        this.raza = raza;
        this.edad = edad;
        this.altura = altura;
        this.sexo = sexo;
    }

    public Perro(String nombre, int edad, float altura, char sexo) {
        this.nombre = nombre;
        this.raza = "Criollo";
        this.edad = edad;
        this.altura = altura;
        this.sexo = sexo;
    }

    public Perro() {
        this.nombre = "";
        this.raza = "";
        this.edad = 0;
        this.sexo = 'I';
        this.altura = 0;
    }

    public void ladrar() {
        System.out.println("Guauuu Guauu");
    }

    public boolean estadoPerro() {
        return this.dormir;
    }

    public void cambiarEstado() {
        if (this.dormir) {
            this.dormir = false;
        } else {
            this.dormir = true;
        }
    }

    @Override
    public String toString() {
        return "Perro [nombre=" + nombre + ", raza=" + raza + ", edad=" + edad + ", altura=" + altura + ", sexo=" + sexo
                + ", dormir=" + dormir + "]";
    }

}

File: Semana2/Python/Perro.py
class Perro:
    nombre = ""
    raza = ""
    edad = 0
    altura = 0.0
    sexo = ''
    dormir = False
    
    def __init__(self, nombre, raza, edad, altura, sexo):
        self.nombre = nombre
        self.raza = raza
        self.edad = edad
        self.altura =  altura
        self.sexo = sexo
        
    def __str__(self) -> str:
        return f"Perro [nombre=,{self.nombre} raza={self.raza} , edad={self.edad} , altura={self.altura}, sexo={self.sexo} , dormir={self.dormir} ]"
    
Firulay = Perro("Firulay",'Pastor Callejero', 4, 0.85,'M')
Firulay.edad = "Cinco"
print(Firulay)